1.30.0

matteobruni opened the pull request:
# tsParticles 1.30.0

## New Features

- Added wobble effect to particles, configurable in options
- Added tilt effect to particles, configurable in options
- Added roll options to particles, configurable in options
- Added all new sections to editor

## Bug Fixes

- Fixed old TypeScript definitions generation, fixes #1683 

## Breaking Changes

- Confetti animations are now split in three generic particles sections (`roll`, `tilt`, `wobble`)
- Removed confetti shape, it's no longer needed
- The library has changed the output folders, the `dist/` folder has been removed.

### A note about the `dist/` folder

#### Imports (`import`)

In most cases this change will affect only who are using internal imports.

__This `tsparticles/dist` can be safely replaced to `tsparticles`__.

#### CDN or Vanilla

This will affect also everyone using tsParticles with CDNs since the `dist` folder is no longer part of the path, but if it's copied from the CDN website it's fine.

### A note about `wobble`, `tilt` and `roll`

These three new particles options sections will create a wonderful and realistic confetti effect to all particles. You can use any `shape` with these effects, even external ones, self-created or not. The only one not compatible is `confetti` because it's been replaced by these properties.

What do they means?

#### Roll

Roll makes particles roll on their `y` axis. They look like they're falling rolling on themself. In these options you can set a back color (that will be visible when the particle has flipped).

The color is optional and a `darken` or a `enlighten` section can be alternatively configured.

The reading values order is:

- Back Color
- Darken
- Enlighten

So if `enlighten` and `darken` are both specified, only `darken` will be read.

`darken` and `enlighten` have a `value` and `enable`, the first will subtract the luminosity of the color for that value, the second will add it, only if the `enable` value is true.

#### Tilt

Tilt is responsible for particles deformation, the particles will be tilted following the specified parameters. This is not animated by default, the options are similar to `rotate`, they don't have `path` since is not valid here.

If a value is set all particles will be tilted for that angle (both horizontally and vertically using `cos` and `sin` functions), when animated the angle will be increased or decreased, depending on set direction.

#### Wobble

Wobble is the horizontal movement that simulates the air resistance. The particles will bounce from left to right with a configurable `speed` and `distance`.

matteobruni committed:
Merge pull request #1663 from matteobruni/staging

1.29.0

matteobruni committed:
feat: added wobble effect to particles, bringing confetti animations to all particles

matteobruni committed:
Merge branch 'main' into wobble_tilt

dependabot[bot] committed:
build(deps-dev): bump eslint from 7.28.0 to 7.29.0

Bumps [eslint](https://github.com/eslint/eslint) from 7.28.0 to 7.29.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/master/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v7.28.0...v7.29.0)

---
updated-dependencies:
- dependency-name: eslint
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @types/chai from 4.2.18 to 4.2.19

Bumps [@types/chai](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/chai) from 4.2.18 to 4.2.19.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/chai)

---
updated-dependencies:
- dependency-name: "@types/chai"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @types/node from 15.12.2 to 15.12.4

Bumps [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) from 15.12.2 to 15.12.4.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

---
updated-dependencies:
- dependency-name: "@types/node"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump ng-packagr from 12.0.5 to 12.0.6

Bumps [ng-packagr](https://github.com/ng-packagr/ng-packagr) from 12.0.5 to 12.0.6.
- [Release notes](https://github.com/ng-packagr/ng-packagr/releases)
- [Changelog](https://github.com/ng-packagr/ng-packagr/blob/master/CHANGELOG.md)
- [Commits](https://github.com/ng-packagr/ng-packagr/compare/v12.0.5...v12.0.6)

---
updated-dependencies:
- dependency-name: ng-packagr
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump core-js from 3.14.0 to 3.15.0

Bumps [core-js](https://github.com/zloirock/core-js/tree/HEAD/packages/core-js) from 3.14.0 to 3.15.0.
- [Release notes](https://github.com/zloirock/core-js/releases)
- [Changelog](https://github.com/zloirock/core-js/blob/master/CHANGELOG.md)
- [Commits](https://github.com/zloirock/core-js/commits/v3.15.0/packages/core-js)

---
updated-dependencies:
- dependency-name: core-js
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1670 from matteobruni/dependabot/npm_and_yarn/dev/core-js-3.15.0

build(deps): bump core-js from 3.14.0 to 3.15.0

matteobruni committed:
Merge pull request #1667 from matteobruni/dependabot/npm_and_yarn/dev/types/chai-4.2.19

build(deps-dev): bump @types/chai from 4.2.18 to 4.2.19

matteobruni committed:
Merge pull request #1666 from matteobruni/dependabot/npm_and_yarn/dev/eslint-7.29.0

build(deps-dev): bump eslint from 7.28.0 to 7.29.0

matteobruni committed:
Merge pull request #1668 from matteobruni/dependabot/npm_and_yarn/dev/types/node-15.12.4

build(deps-dev): bump @types/node from 15.12.2 to 15.12.4

matteobruni committed:
Merge pull request #1669 from matteobruni/dependabot/npm_and_yarn/dev/ng-packagr-12.0.6

build(deps-dev): bump ng-packagr from 12.0.5 to 12.0.6

dependabot[bot] committed:
build(deps-dev): bump mocha from 9.0.0 to 9.0.1

Bumps [mocha](https://github.com/mochajs/mocha) from 9.0.0 to 9.0.1.
- [Release notes](https://github.com/mochajs/mocha/releases)
- [Changelog](https://github.com/mochajs/mocha/blob/master/CHANGELOG.md)
- [Commits](https://github.com/mochajs/mocha/compare/v9.0.0...v9.0.1)

---
updated-dependencies:
- dependency-name: mocha
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump rollup from 2.52.1 to 2.52.2

Bumps [rollup](https://github.com/rollup/rollup) from 2.52.1 to 2.52.2.
- [Release notes](https://github.com/rollup/rollup/releases)
- [Changelog](https://github.com/rollup/rollup/blob/master/CHANGELOG.md)
- [Commits](https://github.com/rollup/rollup/compare/v2.52.1...v2.52.2)

---
updated-dependencies:
- dependency-name: rollup
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @types/react-syntax-highlighter

Bumps [@types/react-syntax-highlighter](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/react-syntax-highlighter) from 13.5.0 to 13.5.1.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/react-syntax-highlighter)

---
updated-dependencies:
- dependency-name: "@types/react-syntax-highlighter"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1671 from matteobruni/dependabot/npm_and_yarn/dev/mocha-9.0.1

build(deps-dev): bump mocha from 9.0.0 to 9.0.1

matteobruni committed:
Merge pull request #1672 from matteobruni/dependabot/npm_and_yarn/dev/rollup-2.52.2

build(deps-dev): bump rollup from 2.52.1 to 2.52.2

matteobruni committed:
Merge pull request #1674 from matteobruni/dependabot/npm_and_yarn/dev/types/react-syntax-highlighter-13.5.1

build(deps-dev): bump @types/react-syntax-highlighter from 13.5.0 to 13.5.1

dependabot[bot] committed:
build(deps-dev): bump svelte-check from 2.1.0 to 2.2.0

Bumps [svelte-check](https://github.com/sveltejs/language-tools) from 2.1.0 to 2.2.0.
- [Release notes](https://github.com/sveltejs/language-tools/releases)
- [Commits](https://github.com/sveltejs/language-tools/compare/svelte-check-2.1.0...svelte-check-2.2.0)

---
updated-dependencies:
- dependency-name: svelte-check
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1673 from matteobruni/dependabot/npm_and_yarn/dev/svelte-check-2.2.0

build(deps-dev): bump svelte-check from 2.1.0 to 2.2.0

matteobruni committed:
feat: added tilt effect to particles, this was called wobble in previous commit
build: disabled wobble effect since it was the tilt effect, this needs new work

matteobruni committed:
build: fixed old ts definitions generation

matteobruni committed:
chore(release): published new version

 - ng-particles@2.12.1
 - inferno-particles@1.12.1
 - jquery-particles@1.29.1
 - preact-particles@1.29.1
 - react-tsparticles@1.29.1
 - riot-particles@1.3.1
 - solid-particles@1.12.1
 - svelte-particles@1.12.1
 - particles.vue@2.12.1
 - particles.vue3@1.12.1
 - web-particles@1.2.1
 - tsparticles-editor@1.29.1
 - tsparticles@1.29.1
 - ng-particles-demo@1.14.1
 - editor-demo@1.14.1
 - inferno-particles-demo@1.12.1
 - jquery-particles-demo@1.14.1
 - preact-particles-demo@1.14.1
 - react-tsparticles-demo@1.14.1
 - riot-particles-demo@1.3.1
 - solid-particles-demo@1.12.1
 - svelte-demo@1.12.1
 - tsparticles-demo@1.14.1
 - particles.vue-demo@1.14.1
 - particles.vue3-demo@1.12.1
 - web-particles-demo@1.14.1
 - tsparticles-preset-big-circles@1.14.1
 - tsparticles-preset-bubbles@1.14.1
 - tsparticles-preset-confetti@1.14.1
 - tsparticles-preset-fire@1.14.1
 - tsparticles-preset-firefly@1.14.1
 - tsparticles-preset-fireworks@1.14.1
 - tsparticles-preset-fountain@1.14.1
 - tsparticles-preset-links@1.14.1
 - tsparticles-preset-sea-anemone@1.14.1
 - tsparticles-preset-snow@1.14.1
 - tsparticles-preset-stars@1.14.1
 - tsparticles-shape-bubble@1.14.1
 - tsparticles-shape-confetti@1.14.1
 - tsparticles-shape-heart@1.14.1
 - tsparticles-shape-multiline-text@1.14.1
 - tsparticles-shape-rounded-rect@1.14.1
 - tsparticles-shape-spiral@1.14.1
 - cra-template-particles-typescript@1.12.1
 - cra-template-particles@1.12.1
 - tsparticles-website@1.29.1

matteobruni committed:
Merge branch 'main' into dev

matteobruni committed:
build: fixed old ts definitions generations

matteobruni committed:
chore(release): published new version

 - ng-particles@2.12.2
 - inferno-particles@1.12.2
 - jquery-particles@1.29.2
 - preact-particles@1.29.2
 - react-tsparticles@1.29.2
 - riot-particles@1.3.2
 - solid-particles@1.12.2
 - svelte-particles@1.12.2
 - particles.vue@2.12.2
 - particles.vue3@1.12.2
 - web-particles@1.2.2
 - tsparticles-editor@1.29.2
 - tsparticles@1.29.2
 - ng-particles-demo@1.14.2
 - editor-demo@1.14.2
 - inferno-particles-demo@1.12.2
 - jquery-particles-demo@1.14.2
 - preact-particles-demo@1.14.2
 - react-tsparticles-demo@1.14.2
 - riot-particles-demo@1.3.2
 - solid-particles-demo@1.12.2
 - svelte-demo@1.12.2
 - tsparticles-demo@1.14.2
 - particles.vue-demo@1.14.2
 - particles.vue3-demo@1.12.2
 - web-particles-demo@1.14.2
 - tsparticles-preset-big-circles@1.14.2
 - tsparticles-preset-bubbles@1.14.2
 - tsparticles-preset-confetti@1.14.2
 - tsparticles-preset-fire@1.14.2
 - tsparticles-preset-firefly@1.14.2
 - tsparticles-preset-fireworks@1.14.2
 - tsparticles-preset-fountain@1.14.2
 - tsparticles-preset-links@1.14.2
 - tsparticles-preset-sea-anemone@1.14.2
 - tsparticles-preset-snow@1.14.2
 - tsparticles-preset-stars@1.14.2
 - tsparticles-shape-bubble@1.14.2
 - tsparticles-shape-confetti@1.14.2
 - tsparticles-shape-heart@1.14.2
 - tsparticles-shape-multiline-text@1.14.2
 - tsparticles-shape-rounded-rect@1.14.2
 - tsparticles-shape-spiral@1.14.2
 - cra-template-particles-typescript@1.12.2
 - cra-template-particles@1.12.2
 - tsparticles-website@1.29.2

matteobruni committed:
Merge pull request #1675 from matteobruni/dev

1.29.2

matteobruni committed:
Merge branch 'main' into wobble_tilt

matteobruni committed:
feat: confetti animations are now splitted in two generic particles sections (tilt, wobble)

matteobruni committed:
build: removed confetti shape, it's no longer needed

matteobruni committed:
feat: added new sections to editor

matteobruni committed:
build: used setTransform instead of transform

matteobruni committed:
Merge pull request #1665 from matteobruni/wobble_tilt

Confetti Effect

matteobruni committed:
build: prettier

dependabot[bot] committed:
build(deps-dev): bump webpack from 5.39.1 to 5.40.0

Bumps [webpack](https://github.com/webpack/webpack) from 5.39.1 to 5.40.0.
- [Release notes](https://github.com/webpack/webpack/releases)
- [Commits](https://github.com/webpack/webpack/compare/v5.39.1...v5.40.0)

---
updated-dependencies:
- dependency-name: webpack
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @typescript-eslint/eslint-plugin

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 4.27.0 to 4.28.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v4.28.0/packages/eslint-plugin)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/eslint-plugin"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @typescript-eslint/parser from 4.27.0 to 4.28.0

Bumps [@typescript-eslint/parser](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/parser) from 4.27.0 to 4.28.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/parser/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v4.28.0/packages/parser)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/parser"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @babel/preset-env from 7.14.5 to 7.14.7

Bumps [@babel/preset-env](https://github.com/babel/babel/tree/HEAD/packages/babel-preset-env) from 7.14.5 to 7.14.7.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.14.7/packages/babel-preset-env)

---
updated-dependencies:
- dependency-name: "@babel/preset-env"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump gh-pages from 3.2.2 to 3.2.3

Bumps [gh-pages](https://github.com/tschaub/gh-pages) from 3.2.2 to 3.2.3.
- [Release notes](https://github.com/tschaub/gh-pages/releases)
- [Changelog](https://github.com/tschaub/gh-pages/blob/main/changelog.md)
- [Commits](https://github.com/tschaub/gh-pages/compare/v3.2.2...v3.2.3)

---
updated-dependencies:
- dependency-name: gh-pages
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1677 from matteobruni/dependabot/npm_and_yarn/dev/webpack-5.40.0

build(deps-dev): bump webpack from 5.39.1 to 5.40.0

matteobruni committed:
Merge pull request #1678 from matteobruni/dependabot/npm_and_yarn/dev/typescript-eslint/eslint-plugin-4.28.0

build(deps-dev): bump @typescript-eslint/eslint-plugin from 4.27.0 to 4.28.0

matteobruni committed:
Merge pull request #1679 from matteobruni/dependabot/npm_and_yarn/dev/typescript-eslint/parser-4.28.0

build(deps-dev): bump @typescript-eslint/parser from 4.27.0 to 4.28.0

matteobruni committed:
Merge pull request #1680 from matteobruni/dependabot/npm_and_yarn/dev/babel/preset-env-7.14.7

build(deps-dev): bump @babel/preset-env from 7.14.5 to 7.14.7

matteobruni committed:
Merge pull request #1681 from matteobruni/dependabot/npm_and_yarn/dev/gh-pages-3.2.3

build(deps): bump gh-pages from 3.2.2 to 3.2.3

matteobruni committed:
fix: fixed #1683
fix: fixed #1682
breaking changes: imports with tsparticles/dist must be renamed to tsparticles

matteobruni committed:
Merge remote-tracking branch 'origin/dev' into dev

ghost commented:
<img src="https://www.deepcode.ai/icons/green_check.svg" width= "50px" align= "left"/> Congratulations :tada:. DeepCode [analyzed](https://www.deepcode.ai/app/gh/matteobruni/tsparticles/551940323d4d4f2619746d22426358837b569164/matteobruni/tsparticles/45602123f0b6462e63245397da63aabb8f75c68c/pr/_/%2F/code/?utm_source=gh_review&c=0&w=0&i=0&) your code in 4.181 seconds and we found no issues. Enjoy a moment of no bugs :sunny:.

#### 👉 View analysis in [**DeepCode’s Dashboard**](https://www.deepcode.ai/app/gh/matteobruni/tsparticles/551940323d4d4f2619746d22426358837b569164/matteobruni/tsparticles/45602123f0b6462e63245397da63aabb8f75c68c/pr/_/%2F/code/?utm_source=gh_review&c=0&w=0&i=0&) | [_Configure the bot_](https://www.deepcode.ai/app/gh/?ownerconfig=matteobruni)
#### 👉 The DeepCode service and API will be deprecated in August, 2021. [Here](https://www.deepcode.ai/blog/migrate-to-snyk-code) is the information how to migrate. Thank you for using DeepCode 🙏 ❤️ !
If you are using our plugins, you might be interested in their successors: [Snyk's JetBrains plugin](https://snyk.co/udpkq) and [Snyk's VS Code plugin](https://snyk.co/udpkr).

dependabot[bot] committed:
build(deps-dev): bump ng-packagr from 12.0.6 to 12.0.8

Bumps [ng-packagr](https://github.com/ng-packagr/ng-packagr) from 12.0.6 to 12.0.8.
- [Release notes](https://github.com/ng-packagr/ng-packagr/releases)
- [Changelog](https://github.com/ng-packagr/ng-packagr/blob/v12.0.8/CHANGELOG.md)
- [Commits](https://github.com/ng-packagr/ng-packagr/compare/v12.0.6...v12.0.8)

---
updated-dependencies:
- dependency-name: ng-packagr
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump svelte from 3.38.2 to 3.38.3

Bumps [svelte](https://github.com/sveltejs/svelte) from 3.38.2 to 3.38.3.
- [Release notes](https://github.com/sveltejs/svelte/releases)
- [Changelog](https://github.com/sveltejs/svelte/blob/master/CHANGELOG.md)
- [Commits](https://github.com/sveltejs/svelte/compare/v3.38.2...v3.38.3)

---
updated-dependencies:
- dependency-name: svelte
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump html-webpack-plugin from 5.3.1 to 5.3.2

Bumps [html-webpack-plugin](https://github.com/jantimon/html-webpack-plugin) from 5.3.1 to 5.3.2.
- [Release notes](https://github.com/jantimon/html-webpack-plugin/releases)
- [Changelog](https://github.com/jantimon/html-webpack-plugin/blob/main/CHANGELOG.md)
- [Commits](https://github.com/jantimon/html-webpack-plugin/commits)

---
updated-dependencies:
- dependency-name: html-webpack-plugin
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump core-js from 3.15.0 to 3.15.1

Bumps [core-js](https://github.com/zloirock/core-js/tree/HEAD/packages/core-js) from 3.15.0 to 3.15.1.
- [Release notes](https://github.com/zloirock/core-js/releases)
- [Changelog](https://github.com/zloirock/core-js/blob/master/CHANGELOG.md)
- [Commits](https://github.com/zloirock/core-js/commits/v3.15.1/packages/core-js)

---
updated-dependencies:
- dependency-name: core-js
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @vue/compiler-sfc from 3.1.1 to 3.1.2

Bumps [@vue/compiler-sfc](https://github.com/vuejs/vue-next/tree/HEAD/packages/compiler-sfc) from 3.1.1 to 3.1.2.
- [Release notes](https://github.com/vuejs/vue-next/releases)
- [Changelog](https://github.com/vuejs/vue-next/blob/master/CHANGELOG.md)
- [Commits](https://github.com/vuejs/vue-next/commits/v3.1.2/packages/compiler-sfc)

---
updated-dependencies:
- dependency-name: "@vue/compiler-sfc"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1684 from matteobruni/dependabot/npm_and_yarn/dev/ng-packagr-12.0.8

build(deps-dev): bump ng-packagr from 12.0.6 to 12.0.8

matteobruni committed:
Merge pull request #1685 from matteobruni/dependabot/npm_and_yarn/dev/svelte-3.38.3

build(deps-dev): bump svelte from 3.38.2 to 3.38.3

matteobruni committed:
Merge pull request #1686 from matteobruni/dependabot/npm_and_yarn/dev/html-webpack-plugin-5.3.2

build(deps-dev): bump html-webpack-plugin from 5.3.1 to 5.3.2

matteobruni committed:
Merge pull request #1687 from matteobruni/dependabot/npm_and_yarn/dev/core-js-3.15.1

build(deps): bump core-js from 3.15.0 to 3.15.1

matteobruni committed:
Merge pull request #1688 from matteobruni/dependabot/npm_and_yarn/dev/vue/compiler-sfc-3.1.2

build(deps-dev): bump @vue/compiler-sfc from 3.1.1 to 3.1.2

matteobruni committed:
build: working on older typescript subfolders imports

matteobruni committed:
Merge remote-tracking branch 'origin/dev' into dev

matteobruni committed:
fix: fixed older typescript version

matteobruni committed:
build: removed all references to old confetti-shape package

matteobruni committed:
build: updated jsdelivr urls

matteobruni committed:
build: improvements to new confetti options

dependabot[bot] committed:
build(deps): bump @angular/router from 12.0.5 to 12.1.0

Bumps [@angular/router](https://github.com/angular/angular/tree/HEAD/packages/router) from 12.0.5 to 12.1.0.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/12.1.0/packages/router)

---
updated-dependencies:
- dependency-name: "@angular/router"
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump @angular/platform-browser from 12.0.5 to 12.1.0

Bumps [@angular/platform-browser](https://github.com/angular/angular/tree/HEAD/packages/platform-browser) from 12.0.5 to 12.1.0.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/12.1.0/packages/platform-browser)

---
updated-dependencies:
- dependency-name: "@angular/platform-browser"
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump @angular/common from 12.0.5 to 12.1.0

Bumps [@angular/common](https://github.com/angular/angular/tree/HEAD/packages/common) from 12.0.5 to 12.1.0.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/12.1.0/packages/common)

---
updated-dependencies:
- dependency-name: "@angular/common"
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1694 from matteobruni/dependabot/npm_and_yarn/dev/angular/router-12.1.0

build(deps): bump @angular/router from 12.0.5 to 12.1.0

matteobruni committed:
Merge pull request #1695 from matteobruni/dependabot/npm_and_yarn/dev/angular/platform-browser-12.1.0

build(deps): bump @angular/platform-browser from 12.0.5 to 12.1.0

matteobruni committed:
Merge pull request #1697 from matteobruni/dependabot/npm_and_yarn/dev/angular/common-12.1.0

build(deps): bump @angular/common from 12.0.5 to 12.1.0

dependabot[bot] committed:
build(deps-dev): bump ng-packagr from 12.0.8 to 12.1.0

Bumps [ng-packagr](https://github.com/ng-packagr/ng-packagr) from 12.0.8 to 12.1.0.
- [Release notes](https://github.com/ng-packagr/ng-packagr/releases)
- [Changelog](https://github.com/ng-packagr/ng-packagr/blob/master/CHANGELOG.md)
- [Commits](https://github.com/ng-packagr/ng-packagr/compare/v12.0.8...v12.1.0)

---
updated-dependencies:
- dependency-name: ng-packagr
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump @angular/forms from 12.0.5 to 12.1.0

Bumps [@angular/forms](https://github.com/angular/angular/tree/HEAD/packages/forms) from 12.0.5 to 12.1.0.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/12.1.0/packages/forms)

---
updated-dependencies:
- dependency-name: "@angular/forms"
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @angular-devkit/build-angular

Bumps [@angular-devkit/build-angular](https://github.com/angular/angular-cli) from 12.0.5 to 12.1.0.
- [Release notes](https://github.com/angular/angular-cli/releases)
- [Commits](https://github.com/angular/angular-cli/compare/v12.0.5...v12.1.0)

---
updated-dependencies:
- dependency-name: "@angular-devkit/build-angular"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump typedoc from 0.21.0 to 0.21.1

Bumps [typedoc](https://github.com/TypeStrong/TypeDoc) from 0.21.0 to 0.21.1.
- [Release notes](https://github.com/TypeStrong/TypeDoc/releases)
- [Commits](https://github.com/TypeStrong/TypeDoc/compare/v0.21.0...v0.21.1)

---
updated-dependencies:
- dependency-name: typedoc
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1699 from matteobruni/dependabot/npm_and_yarn/dev/ng-packagr-12.1.0

build(deps-dev): bump ng-packagr from 12.0.8 to 12.1.0

matteobruni committed:
Merge pull request #1698 from matteobruni/dependabot/npm_and_yarn/dev/angular/forms-12.1.0

build(deps): bump @angular/forms from 12.0.5 to 12.1.0

matteobruni committed:
Merge pull request #1696 from matteobruni/dependabot/npm_and_yarn/dev/angular-devkit/build-angular-12.1.0

build(deps-dev): bump @angular-devkit/build-angular from 12.0.5 to 12.1.0

matteobruni committed:
Merge pull request #1701 from matteobruni/dependabot/npm_and_yarn/dev/typedoc-0.21.1

build(deps-dev): bump typedoc from 0.21.0 to 0.21.1

dependabot[bot] committed:
build(deps-dev): bump style-loader from 2.0.0 to 3.0.0

Bumps [style-loader](https://github.com/webpack-contrib/style-loader) from 2.0.0 to 3.0.0.
- [Release notes](https://github.com/webpack-contrib/style-loader/releases)
- [Changelog](https://github.com/webpack-contrib/style-loader/blob/master/CHANGELOG.md)
- [Commits](https://github.com/webpack-contrib/style-loader/compare/v2.0.0...v3.0.0)

---
updated-dependencies:
- dependency-name: style-loader
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @angular/cli from 12.0.5 to 12.1.0

Bumps [@angular/cli](https://github.com/angular/angular-cli) from 12.0.5 to 12.1.0.
- [Release notes](https://github.com/angular/angular-cli/releases)
- [Commits](https://github.com/angular/angular-cli/compare/v12.0.5...v12.1.0)

---
updated-dependencies:
- dependency-name: "@angular/cli"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1700 from matteobruni/dependabot/npm_and_yarn/dev/angular/cli-12.1.0

build(deps-dev): bump @angular/cli from 12.0.5 to 12.1.0

matteobruni committed:
Merge pull request #1703 from matteobruni/dependabot/npm_and_yarn/dev/style-loader-3.0.0

build(deps-dev): bump style-loader from 2.0.0 to 3.0.0

dependabot[bot] committed:
build(deps): bump @angular/animations from 12.0.5 to 12.1.0

Bumps [@angular/animations](https://github.com/angular/angular/tree/HEAD/packages/animations) from 12.0.5 to 12.1.0.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/12.1.0/packages/animations)

---
updated-dependencies:
- dependency-name: "@angular/animations"
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump @angular/platform-browser-dynamic

Bumps [@angular/platform-browser-dynamic](https://github.com/angular/angular/tree/HEAD/packages/platform-browser-dynamic) from 12.0.5 to 12.1.0.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/12.1.0/packages/platform-browser-dynamic)

---
updated-dependencies:
- dependency-name: "@angular/platform-browser-dynamic"
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump @angular/core from 12.0.5 to 12.1.0

Bumps [@angular/core](https://github.com/angular/angular/tree/HEAD/packages/core) from 12.0.5 to 12.1.0.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/12.1.0/packages/core)

---
updated-dependencies:
- dependency-name: "@angular/core"
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1704 from matteobruni/dependabot/npm_and_yarn/dev/angular/platform-browser-dynamic-12.1.0

build(deps): bump @angular/platform-browser-dynamic from 12.0.5 to 12.1.0

matteobruni committed:
Merge pull request #1702 from matteobruni/dependabot/npm_and_yarn/dev/angular/animations-12.1.0

build(deps): bump @angular/animations from 12.0.5 to 12.1.0

matteobruni committed:
Merge pull request #1705 from matteobruni/dependabot/npm_and_yarn/dev/angular/core-12.1.0

build(deps): bump @angular/core from 12.0.5 to 12.1.0

matteobruni committed:
build: improvements to new confetti options

matteobruni committed:
build: fixed the distfiles script

dependabot[bot] committed:
build(deps-dev): bump prettier from 2.3.1 to 2.3.2

Bumps [prettier](https://github.com/prettier/prettier) from 2.3.1 to 2.3.2.
- [Release notes](https://github.com/prettier/prettier/releases)
- [Changelog](https://github.com/prettier/prettier/blob/main/CHANGELOG.md)
- [Commits](https://github.com/prettier/prettier/compare/2.3.1...2.3.2)

---
updated-dependencies:
- dependency-name: prettier
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump node-sass from 6.0.0 to 6.0.1

Bumps [node-sass](https://github.com/sass/node-sass) from 6.0.0 to 6.0.1.
- [Release notes](https://github.com/sass/node-sass/releases)
- [Changelog](https://github.com/sass/node-sass/blob/master/CHANGELOG.md)
- [Commits](https://github.com/sass/node-sass/compare/v6.0.0...v6.0.1)

---
updated-dependencies:
- dependency-name: node-sass
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @types/node from 15.12.4 to 15.12.5

Bumps [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) from 15.12.4 to 15.12.5.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

---
updated-dependencies:
- dependency-name: "@types/node"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump uglify-js from 3.13.9 to 3.13.10

Bumps [uglify-js](https://github.com/mishoo/UglifyJS) from 3.13.9 to 3.13.10.
- [Release notes](https://github.com/mishoo/UglifyJS/releases)
- [Commits](https://github.com/mishoo/UglifyJS/compare/v3.13.9...v3.13.10)

---
updated-dependencies:
- dependency-name: uglify-js
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump eslint-plugin-vue from 7.11.1 to 7.12.1

Bumps [eslint-plugin-vue](https://github.com/vuejs/eslint-plugin-vue) from 7.11.1 to 7.12.1.
- [Release notes](https://github.com/vuejs/eslint-plugin-vue/releases)
- [Commits](https://github.com/vuejs/eslint-plugin-vue/compare/v7.11.1...v7.12.1)

---
updated-dependencies:
- dependency-name: eslint-plugin-vue
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1710 from matteobruni/dependabot/npm_and_yarn/dev/prettier-2.3.2

build(deps-dev): bump prettier from 2.3.1 to 2.3.2

matteobruni committed:
Merge pull request #1711 from matteobruni/dependabot/npm_and_yarn/dev/node-sass-6.0.1

build(deps-dev): bump node-sass from 6.0.0 to 6.0.1

matteobruni committed:
Merge pull request #1712 from matteobruni/dependabot/npm_and_yarn/dev/types/node-15.12.5

build(deps-dev): bump @types/node from 15.12.4 to 15.12.5

dependabot[bot] committed:
build(deps-dev): bump babel-preset-solid from 0.26.5 to 1.0.0

Bumps [babel-preset-solid](https://github.com/solidjs/solid) from 0.26.5 to 1.0.0.
- [Release notes](https://github.com/solidjs/solid/releases)
- [Changelog](https://github.com/solidjs/solid/blob/main/CHANGELOG.md)
- [Commits](https://github.com/solidjs/solid/compare/v0.26.5...v1.0.0)

---
updated-dependencies:
- dependency-name: babel-preset-solid
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1713 from matteobruni/dependabot/npm_and_yarn/dev/uglify-js-3.13.10

build(deps-dev): bump uglify-js from 3.13.9 to 3.13.10

matteobruni committed:
Merge pull request #1714 from matteobruni/dependabot/npm_and_yarn/dev/eslint-plugin-vue-7.12.1

build(deps-dev): bump eslint-plugin-vue from 7.11.1 to 7.12.1

dependabot[bot] committed:
build(deps-dev): bump typedoc from 0.21.1 to 0.21.2

Bumps [typedoc](https://github.com/TypeStrong/TypeDoc) from 0.21.1 to 0.21.2.
- [Release notes](https://github.com/TypeStrong/TypeDoc/releases)
- [Commits](https://github.com/TypeStrong/TypeDoc/compare/v0.21.1...v0.21.2)

---
updated-dependencies:
- dependency-name: typedoc
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump rollup from 2.52.2 to 2.52.3

Bumps [rollup](https://github.com/rollup/rollup) from 2.52.2 to 2.52.3.
- [Release notes](https://github.com/rollup/rollup/releases)
- [Changelog](https://github.com/rollup/rollup/blob/master/CHANGELOG.md)
- [Commits](https://github.com/rollup/rollup/compare/v2.52.2...v2.52.3)

---
updated-dependencies:
- dependency-name: rollup
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump terser-webpack-plugin from 5.1.3 to 5.1.4

Bumps [terser-webpack-plugin](https://github.com/webpack-contrib/terser-webpack-plugin) from 5.1.3 to 5.1.4.
- [Release notes](https://github.com/webpack-contrib/terser-webpack-plugin/releases)
- [Changelog](https://github.com/webpack-contrib/terser-webpack-plugin/blob/master/CHANGELOG.md)
- [Commits](https://github.com/webpack-contrib/terser-webpack-plugin/compare/v5.1.3...v5.1.4)

---
updated-dependencies:
- dependency-name: terser-webpack-plugin
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1715 from matteobruni/dependabot/npm_and_yarn/dev/babel-preset-solid-1.0.0

build(deps-dev): bump babel-preset-solid from 0.26.5 to 1.0.0

matteobruni committed:
Merge pull request #1716 from matteobruni/dependabot/npm_and_yarn/dev/typedoc-0.21.2

build(deps-dev): bump typedoc from 0.21.1 to 0.21.2

matteobruni committed:
Merge pull request #1717 from matteobruni/dependabot/npm_and_yarn/dev/rollup-2.52.3

build(deps-dev): bump rollup from 2.52.2 to 2.52.3

matteobruni committed:
Merge pull request #1718 from matteobruni/dependabot/npm_and_yarn/dev/terser-webpack-plugin-5.1.4

build(deps-dev): bump terser-webpack-plugin from 5.1.3 to 5.1.4

matteobruni committed:
feat: added roll options to particles, this is the latest missing feature to the confetti rework

matteobruni committed:
fix: fixed confetti preset, added roll options

matteobruni committed:
feat: added particle back color used in roll options

matteobruni committed:
fix: fixed backcolor when undefined

matteobruni committed:
feat: added roll editor
